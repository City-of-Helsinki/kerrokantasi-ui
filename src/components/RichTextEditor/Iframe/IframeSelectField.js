import React from 'react';
import PropTypes from 'prop-types';

function IframeSelectField(props) {
  const {name, label, handleInputChange, value, options, isRequired, errorMsg} = props;
  const errorId = `iframe-input-error-${name}`;

  // options = [{value: "value", text: "text"}]
  const optionElements = options.map(option => {
    return (
      <option value={option.value} key={option.value}>{option.text}</option>
    );
  });

  return (
    <div className="input-container">
      <label htmlFor={`iframe-${name}`}>{label} {isRequired && '*'}</label>
      <select
        id={`iframe-${name}`}
        className="rich-text-editor-form-select-input"
        name={name}
        value={value}
        onChange={handleInputChange}
      >
        {optionElements}
      </select>
      {errorMsg && <p id={errorId} role="alert" className="iframe-input-error">{errorMsg}</p>}
    </div>
  );
}

IframeSelectField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  isRequired: PropTypes.bool,
  errorMsg: PropTypes.string,
};

export default IframeSelectField;
