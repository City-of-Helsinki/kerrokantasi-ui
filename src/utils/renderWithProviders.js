/* eslint react/prop-types: 0 */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';
import { IntlProvider } from 'react-intl';

import createStore from "../createStore";
import { userOidcConfig, apiTokenClientConfig } from './oidcConfig';

const loginProviderConfig = {
  userManagerSettings: userOidcConfig,
  apiTokensClientSettings: apiTokenClientConfig,
  sessionPollerSettings: { pollIntervalInMs: 10000 },
}

const renderWithProviders = (
  ui,
  { preloadedState = {}, locale = 'fi', history, store = createStore(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider {...loginProviderConfig}>
      <IntlProvider locale={locale}>
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </LoginProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};


export default renderWithProviders;
