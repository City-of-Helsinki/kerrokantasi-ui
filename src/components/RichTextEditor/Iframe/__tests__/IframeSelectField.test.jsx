import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import IframeSelectField from '../IframeSelectField';
import renderWithProviders from '../../../../utils/renderWithProviders';

const renderComponent = (propOverrides) => {
  const props = {
    name: 'test-name',
    label: 'test-label',
    handleInputChange: jest.fn(),
    value: 'test-value',
    options: [
      { value: 'no', text: 'text-no' },
      { value: 'yes', text: 'text-yes' },
    ],
    ...propOverrides,
  };

  return renderWithProviders(
    <MemoryRouter>
      <IframeSelectField {...props} />
    </MemoryRouter>,
  );
};

describe('<IframeSelectField />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('should handle input onChange', async () => {
    const handleInputChangeMock = jest.fn();

    renderComponent({ handleInputChange: handleInputChangeMock });

    const select = await screen.findByRole('combobox');
    const option = await screen.findByRole('option', { name: 'text-no' });

    const user = userEvent.setup();

    await user.selectOptions(select, option);

    await waitFor(() => expect(handleInputChangeMock).toHaveBeenCalled());
  });
});
