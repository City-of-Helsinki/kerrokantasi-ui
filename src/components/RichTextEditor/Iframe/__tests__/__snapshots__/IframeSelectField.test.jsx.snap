// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IframeSelectField /> should render correctly 1`] = `
<div>
  <div
    class="input-container"
  >
    <label
      for="iframe-test-name"
    >
      test-label
       
    </label>
    <select
      class="rich-text-editor-form-select-input"
      id="iframe-test-name"
      name="test-name"
    >
      <option
        value="no"
      >
        text-no
      </option>
      <option
        value="yes"
      >
        text-yes
      </option>
    </select>
  </div>
</div>
`;
